syntax = "proto3";

option optimize_for = SPEED;

option cc_enable_arenas = true;

package NVcs;

import "aapi/lib/proto/types.proto";

service Vcs {
/// ========== proxy <-> client protocol

    /// Returns current svn head revision.
    rpc SvnHead(NAapi.TEmpty) returns (NAapi.TRevision);

    /// Returns 20-bytes hg changeset for a given name
    rpc HgId(NAapi.TName) returns (NAapi.THash);

    rpc Objects2(stream NAapi.THash) returns (stream NAapi.TObject);
    rpc Objects3(stream NAapi.THashes) returns (stream NAapi.TObjects);

    rpc Walk(NAapi.THash) returns (stream NAapi.TDirectories);

//    rpc Diff(NAapi.TDiffRequest) returns (stream NAapi.TDiffElement);

/// ========== proxy <-> converter protocol

    /// Puts provided objects into local cache.
    rpc Push(stream NAapi.TObjects) returns (NAapi.TEmpty);

    /// Notifies proxy abuot new objects in YT table.
    rpc NotifyNewObjects(stream NAapi.THashes) returns (NAapi.TEmpty);

}
